examples/astro/src/react/content-types/calendarEvent.tsx

```diff
@@ -0,0 +1,99 @@
+import type { FC } from "react";
+import type { DotCMSContentlet } from "../../types";
+import useImageSrc from "../hooks/useImageSrc";
+
+export type CalendarEventProps = DotCMSContentlet;
+
+export type LocationsAndActivities = {
+  locations: any[];
+  activities: any[];
+};
+
+const extractLocationsAndActivities = (
+  contentlet: DotCMSContentlet,
+): LocationsAndActivities => {
+  return (
+    contentlet?.reduce(
+      (acc: any, { activities, ...location }: LocationsAndActivities) => {
+        acc.activities = acc.activities.concat(activities);
+        acc.locations.push(location);
+
+        return acc;
+      },
+      { locations: [], activities: [] },
+    ) ?? {
+      locations: [],
+      activities: [],
+    }
+  );
+};
```

Comment:
This was breaking somehow in the nextjs example, how can we make sure is not broken here?

Reply:
I added the optional chaining operator